changedetectionio/fetch_site_status.py

```diff
@@ -276,11 +277,28 @@ def run(self, uuid):
                 else:
                     logging.debug("check_unique_lines: UUID {} had unique content".format(uuid))
 
+            diff_filters = {
+                "add": watch.get('trigger_on_add', True),
+                "del": watch.get('trigger_on_del', True),
+            }
+
+            if False in diff_filters.values(): # if we are supposed to filter any diff types
+                print(">>>>>>>>>>>> Filtering diffs")
+                # get the diff types present in the watch
+                diff_types = watch.get_diff_types(str(stripped_text_from_html))
+                
+                # for each diff type, if the filter setting is false, and the diff type is present, then set the changed_detected to false
+                for diff_type in diff_types:
+                    if not diff_filters[diff_type] and diff_types[diff_type]:
+                        changed_detected = False
+                        break # we only need to check one diff type
+
         # Always record the new checksum
         update_obj["previous_md5"] = fetched_md5
 
         # On the first run of a site, watch['previous_md5'] will be None, set it the current one.
         if not watch.get('previous_md5'):
             watch['previous_md5'] = fetched_md5
+            watch['previous_text'] = str(stripped_text_from_html)
```

Comment:
I really dont like this, better would be to use the `.history` and examine the previous one if theres some special configuration that is enabled.. thoughts?